components/bl-weather-component/component.json

```diff
@@ -0,0 +1,185 @@
+      "settingTooltip": "Select to display the weather for your location."
+    },
+    {
+      "type": "text",
+      "name": "city",
+      "showInSettings": true,
+      "defaultValue": "New York",
+      "hasLogicHandler": false,
+      "handlerId": "cityLogic",
+      "label": "City",
+      "settingTooltip": "City to display the weather for it."
+    },
+    {
+      "type": "json",
+      "name": "coords",
+      "showInSettings": true,
+      "defaultValue": "{\"lat\": 40.730610, \"lng\": -73.935242}",
+      "hasLogicHandler": true,
+      "handlerId": "coordsLogic",
+      "settingTooltip": "Latitude and longitude coordinates to display the weather for them.",
+      "handlerDescription": "This is a handler for the logic to determine the latitude and longitude coordinates to display the weather for them.",
+      "label": "Coordinates",
+      "handlerLabel": "Coordinates Logic",
+      "dataBinding": true
+    },
+    {
+      "type": "checkbox",
+      "name": "forecast",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "forecastLogic",
+      "settingTooltip": "Select to display the weather forecast.",
+      "label": "Forecast"
+    },
+    {
+      "type": "select",
+      "name": "unit",
+      "showInSettings": true,
+      "defaultValue": "standard",
+      "hasLogicHandler": false,
+      "handlerId": "unitLogic",
+      "options": [
+        {
+          "value": "standard",
+          "label": "standard"
+        },
+        {
+          "value": "metric",
+          "label": "metric"
+        },
+        {
+          "value": "imperial",
+          "label": "imperial"
+        }
+      ],
+      "settingTooltip": "Choose a unit of measurement.",
+      "label": "Unit"
+    },
+    {
+      "type": "json",
+      "name": "theme",
+      "showInSettings": true,
+      "defaultValue": "{\r\n  \"fontFamily\": \"Helvetica, sans-serif\",\r\n  \"gradientStart\": \"#0181C2\",\r\n  \"gradientMid\": \"#04A7F9\",\r\n  \"gradientEnd\": \"#4BC4F7\",\r\n  \"locationFontColor\": \"#FFF\",\r\n  \"todayTempFontColor\": \"#FFF\",\r\n  \"todayDateFontColor\": \"#B5DEF4\",\r\n  \"todayRangeFontColor\": \"#B5DEF4\",\r\n  \"todayDescFontColor\": \"#B5DEF4\",\r\n  \"todayInfoFontColor\": \"#B5DEF4\",\r\n  \"todayIconColor\": \"#FFF\",\r\n  \"forecastBackgroundColor\": \"#FFF\",\r\n  \"forecastSeparatorColor\": \"#DDD\",\r\n  \"forecastDateColor\": \"#777\",\r\n  \"forecastDescColor\": \"#777\",\r\n  \"forecastRangeColor\": \"#777\",\r\n  \"forecastIconColor\": \"#4BC4F7\",\r\n  \"containerDropShadow\": \"0px 3px 6px 0px rgba(50, 50, 50, 0.5)\"\r\n}",
+      "hasLogicHandler": true,
+      "handlerId": "themeLogic",
+      "label": "Custom styling",
+      "handlerLabel": "Custom Styling Logic",
+      "handlerDescription": "This is a handler for the logic to determine the custom styling options for the component.",
+      "settingTooltip": "Custom styling options for the component."
+    },
```

Comment:
Could we add variables for this properties?

Reply:
This was discussed with Vova, we decided to add it as a separate parameter, because there is support in the used library.